Automattic/wp-calypso
# Pages: set homepage from Calypso

This pull request makes it possible to set the homepage for a site from Calypso Pages. It is part of a larger epic and is behind a feature flag, `manage/pages/set-homepage`, that is only enabled in `development`.

An option to set a static page to the homepage for a site is available in the popover menu:

<img width="719" alt="screencapture at wed oct 12 12 52 45 edt 2016" src="https://cloud.githubusercontent.com/assets/2098816/19319593/8231fbac-907b-11e6-8749-6566731ca1e2.png">

If a page is already set as the homepage, it will be reflected in the menu:

<img width="735" alt="screencapture at wed oct 12 12 53 06 edt 2016" src="https://cloud.githubusercontent.com/assets/2098816/19319594/82568ab2-907b-11e6-936c-0dac3be97b7f.png">

The 'Blog Posts' page placeholder will be updated appropriately if a page is set as the homepage.

Note: There is currently an issue where cached site options are used when reloading Calypso. So, in order to see the correct homepage setting in the Pages screen, you need to reload Calypso twice. This will be address in a future PR prior to this epic being launched.

To test:
- Try changing the home page using the popover menu for a static page and make sure it changes. Verify in wp-admin and by viewing the site.
- Change things back to the default (blog posts instead of a static page as the homepage) in wp-admin and make sure Calypso reflects that after a page reload.
- Disable the feature flag and make sure thing behave as previously (`DISABLE_FEATURES=manage/pages/set-homepage make run`)
- Run the unit tests for the Redux code and make sure things pass:

```
npm run test-client -- --grep "state sites"
npm run test-client -- --grep "state pages"
```
